tests/trestle/core/commands/validate_test.py

```diff
@@ -284,3 +291,113 @@ def test_validate_catalog_params(sample_catalog_rich_controls: Catalog) -> None:
     param_0_id = sample_catalog_rich_controls.groups[0].controls[0].params[0].id
     sample_catalog_rich_controls.groups[0].controls[0].params[1].id = param_0_id
     assert not validator.model_is_valid(sample_catalog_rich_controls, False)
+
+
+def test_rules_validator_unhappy(tmp_trestle_dir: pathlib.Path) -> None:
+    """Test validation of rules in SSP."""
+    args = argparse.Namespace(mode=const.VAL_MODE_RULES)
+    validator: Validator = validator_factory.get(args)
+    prof_name = 'comp_prof'
+    ssp_name = 'my_ssp'
+
+    gen_args, _ = setup_for_ssp(tmp_trestle_dir, prof_name, ssp_name)
+    args_compdefs = gen_args.compdefs
+
+    # first create the markdown
+    ssp_gen = SSPGenerate()
+    assert ssp_gen._run(gen_args) == 0
+
+    # nassemble the ssp
+    ssp_assemble = SSPAssemble()
+    args = argparse.Namespace(
+        trestle_root=tmp_trestle_dir,
+        markdown=ssp_name,
+        output=ssp_name,
+        verbose=0,
+        regenerate=False,
+        version=None,
+        name=None,
+        compdefs=args_compdefs
+    )
+    assert ssp_assemble._run(args) == 0
+
+    orig_ssp, _ = ModelUtils.load_model_for_class(tmp_trestle_dir, ssp_name, ossp.SystemSecurityPlan)
+
+    by_components = orig_ssp.control_implementation.implemented_requirements[1].by_components
+    set_parameter = orig_ssp.control_implementation.implemented_requirements[0].by_components[0].set_parameters[1]
+    props = orig_ssp.control_implementation.implemented_requirements[0].by_components[0].props
+    by_components[0].props = props
+    by_components[0].set_parameters = []
+    by_components[0].set_parameters.append(set_parameter)
+    by_components[0].set_parameters[0].values.append('shared_param_1_ab_opt_2')
```

Comment:
I believe this is passing a reference rather than a copy.  As a result you are changing the values in both places and the result will be valid - which is not what you want.  I think making a copy of the set_param that you are adding will fix it - since the test is failing because the model is showing as valid.

Reply:
A short comment describing what you are doing would be good.  It should be easy to make it clear that you are copying a setparam fron one bycomp so you can then set its value to something different.

Reply:
Got it, addressed